### ISSUE TITLE
Include bit depth in returns for scipy.io.wavfile.read()

### ISSUE THREAD

**Issue by flueterflam** [scipy/scipy#7244]
The function scipy.io.wavfile.read() returns the sample rate and data from a `.wav` file.  I contend that the bit depth is also important and necessary in some circumstances.  It's trivial, in fact, to add the bit depth to the returns for this function, because it is already being utilized in the function.

Simply change `return fs, data` to `return fs, bit_depth, data`
(several versions of the 24-bit accessible version from [https://github.com/scipy/scipy/issues/6849](https://github.com/scipy/scipy/issues/6849) seem to exist, and depending on the version, add `bit_depth` accordingly, if it is ever committed to master branch)

My concern is that this will cause compatibility issues.  Therefore, I am opening up this issue to see if this change is a) reasonable and b) there are approaches to mitigating potential problems.

---
_Problem:_
If you read a `.wav` file and need the bit depth of it, the current `read()` function requires that you need to open the file again via another method to get the bit depth.  Secondly, bit depth is already used in `read()`, so the extra call could be unnecessary if incorporated into the return values.

_Reason to change:_
The bit depth may not be important for many users of the library.  However, if you create spectrograms from `.wav` data, you will find that the resolution of the plots is different.  This is not immediately scipy.io.wavfile's problem...but if a user wants to create an algorithm to deal with normalizing the resolution, they need access to that bit depth (which scipy.io.wavfile.read() doesn't provide...but could).

Now, one or two extra reads (via a different library or method) may not be so bad.  Once you start getting large `.wav` files and/or a lot of them in your Python code, this can slow things down dramatically.  It is also requiring you to opening double the input file streams necessary.

_Example:_
Here is a MWE that I used to create an image for 16-bit unsigned (PCM), 24-bit unsigned (PCM) (see [gh-6849](https://github.com/scipy/scipy/issues/6849)), and 32-bit unsigned (PCM) `.wav` files.  "test" was recorded in Audacity (as 32-bit float, 192kHz, mono) and exported into each of the respective bit depth `.wav` files.

	import matplotlib
	import matplotlib.pyplot as plt
	import sys
	#Current
	import scipy.io.wavfile
	#24bit accessible version
	sys.path.append('./')
	import wavfile

	def create_spectrogram_from_wav_file():

		#Current scipy.io.wavfile
		#samplerate, data = scipy.io.wavfile.read('test_16bit_signed.wav')
		#samplerate, data = scipy.io.wavfile.read('test_32bit_signed.wav')
		#24-bit accessible scipy.io.wavfile
		samplerate, data = wavfile.read('test_24bit_signed.wav')

		#maximum frequency
		Fs = 10000.

		Pxx, freqs, spectimes, cax = plt.specgram(data, Fs=samplerate)

		plt.show()

	create_spectrogram_from_wav_file()

As you can see from the attached `.png` files, the resolution gets better and better as you increase the bit depth.

For users familiar with Praat or other spectrogram viewers (you can use Audacity also, for example), you will see that that the spectrograms are largely identical for all three `.wav` files.  (In Audiactiy, the resolution is identical but brightness is lower for lower bit depths and higher for higher bit depths...not sure why).  Yet, no such automatic normalization is available via `matplotlib.pyplot.specgram` (that I am aware of ...I searched, really).

Manually normalizing is possible by means of the `NFFT` parameter of `matplotlib.pyplot.specgram`.  For example, for 16-bit you can set `NFFT` to `256`, for 24-bit to `512`, and 32-bit to `768`.  Full circle - here we see how easy it is to normalize with the bit depth that can be easily passed to us via `scipy.io.wavfile.read()`.

Attached are the `.png` files generated from the spectrograms in the MWE and the `.wav` files (which were zipped to allow attachment here) I used to make the spectrogram.

ZIP of `.wav` files: [16+24+32-bit_wav.zip](https://github.com/scipy/scipy/files/884464/16.24.32-bit_wav.zip)

_16-bit `.wav` spectrogram:_
![16bit_signed](https://cloud.githubusercontent.com/assets/5010253/24535539/7662c39e-1620-11e7-855c-fcbae0cf14ab.png)

_24-bit `.wav` spectrogram:_
![24bit_signed](https://cloud.githubusercontent.com/assets/5010253/24535538/7658e676-1620-11e7-993c-854469c3835f.png)

_32-bit `.wav` spectrogram:_
![32bit_signed](https://cloud.githubusercontent.com/assets/5010253/24535540/766b7624-1620-11e7-8dde-43eef6b7bbc1.png)

**Comment by Eric89GXL** [comment]
A `return_bitrate=False` kwarg seems like a reasonable approach to me, but let's wait to see if others agree before you bother implementing. cc @e-q @endolith

**Comment by perimosocordiae** [comment]
The in-progress PR gh-6852 addresses this (I think) by including bit depth in an optional metadata dictionary. Can you try out that branch and see if it solves your issue?

**Comment by flueterflam** [comment]
@perimosocordiae ah, yes, it does.  Actually, I had looked at that issue before.  I didn't link it above because I skimmed the comments at the beginning of the `read()` function and it only listed the current standard returns (sample rate and data).  To avoid confusion, it would be helpful if those comments to could be updated to reflect the changes you made in the function itself.

**Comment by e-q** [comment]
Returning bit depth sounds desirable, I don't have much to add to that. 

However, on your comments regarding normalization: The color scales you are getting from matplotlib are some auto-scaling result based on the outliers of the data set, and aren't invariant when being run on the different files since their numerical amplitudes are very different. (i.e. If you plot color bars on your examples above, you'll see very different scales on the different plots.) Looking at the files as raw time series, one thing you could try is making the 16 and 32 bit series roughly the same amplitude by multiplying the 16-bit data by `2**16`. 

Changing the FFT length via `NFFT` is changing the resolution of the spectrogram, which may or may not be what you want. For clarity in your code, it may be helpful to use `scipy.signal.spectrogram` to calculate the spectrogram values themselves and handle plotting and color mapping separately with matplotlib commands, to make your calculations and their presentation more independent.

**Comment by flueterflam** [comment]
@e-q Yes, I realized from your comment that I rushed through explaining that section.  In my actual code, I do multiply the data amplitude for normalization.  I still needed `NFFT` to normalize the resolution as well, for some reason.  For both of them, I needed to know the bit depth for use in the multiplier.
